### PR TITLE
NoOp option for for TT

### PR DESCRIPTION
**Pull request by mikedodge04**
NoOp option for for TT

### CODE REVIEW

File: lib/taste_tester/host.rb [hunk 1]
```diff
@@ -72,8 +73,10 @@ def runchef
     end
 
     def get_transport
-      if TasteTester::Config.locallink
+      if TasteTester::Config.transport.casecmp('locallink').zero?
```
Comment (jaymzh) [inline lib/taste_tester/host.rb]: probably easier to do:

```
case TasteTester::Config.transport
when 'locallink'
  ...
when 'noop'
  ...
else
   ...
end
```

File: lib/taste_tester/noop.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    end
+
+    alias << add
+
+    def run
+      @status, @output = exec(cmd, logger)
+    end
+
+    def run!
+      @status, @output = exec!(cmd, logger)
```
Comment (jaymzh) [inline lib/taste_tester/noop.rb]: The whole point of no-op is to NOT RUN anything on the remote host. Just don't do anything.

**Comment by jaymzh** [comment]
This looks good to me. Did you test it at all?

**Comment by mikedodge04** [comment]
No I will test in the morning.

File: lib/taste_tester/noop.rb [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      [0, "# TasteTester by #{@user}"]
+    end
+
+    private
+
+    def cmd
+      @cmds.each do |cmd|
+        logger.info("No-op, faking run of: '#{cmd}' on #{@host}")
+      end
+      cmds = @cmds.join(' && ')
```
Comment (jaymzh) [inline lib/taste_tester/noop.rb]: You can drop the rest of this. It doesn't matter, just as long as we have a track record of what we're not doing in debug mode.

File: lib/taste_tester/host.rb [hunk 4]
```diff
@@ -71,13 +72,22 @@ def runchef
       end
     end
 
+    def print_noop_warning
+      @PRINTED_WARNING ||= logger.warn(
```
Comment (jaymzh) [inline lib/taste_tester/host.rb]: You can do that, but it's easier to do it the initializer for NoOp, and use a class variable (@@)

File: lib/taste_tester/noop.rb [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    end
+
+    def print_noop_warning
+      @@PRINTED_WARNING ||= logger.warn(
+        'No-op plugin active, no remote commands will be run!'
+      )
+    end
+
+    def run!
+      print_noop_warning
```
Comment (jaymzh) [inline lib/taste_tester/noop.rb]: I'd move this to the initializer. And add comment on why it's a class var and not an instance var

**Pull request merged**